### PR TITLE
Added custom tables page to user's guide

### PR DESCRIPTION
**Pull request by spficklin**
# Documentation
Issue #74 

## Description

We did not have a page for working with Custom tables.  This issue has sat in the queue for years and it was easy enough to add the docs.  The issue also mentions we are missing the materialized views page but we have since added that. 

## Testing?

Pull the branch, build the docs and view it!

## Screenshots (if appropriate):
custom_tables 1
custom_tables 2
custom_tables 3

### CODE REVIEW

File: docs/user_guide/custom_tables.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Custom Tables
+=============
+
+Chado has a large number of tables that can be used for storing a variety of biological and ancillary data.  All effort should be made to use the existing tables for storing data. However, there are times when the current set of tables is not adequate for storing some data. Tripal allows you to create custom tables in Chado using the Tripal web interface.  The interface for managing custom tables can be found on the **Tripal → Data Storage → Chado -> Custom Tables** page.
+
+.. image:: ./custom_tables.1.png
+
+This page contains a link to add a new custom table (at the top), a search interface to find tables by name, and a listing of existing tables.  You can use this interface to add, edit and remove custom tables.  By defaault, several custom tables have been created automatically by Tripal modules.  These tables hold data managed by the modules.  Also, tables used as materialized views appear in this list as well. They are identified with thee `Is Mview` column.
```
Comment (bradfordcondon) [inline docs/user_guide/custom_tables.rst]:
```suggestion
This page contains a link to add a new custom table (at the top), a search interface to find tables by name, and a listing of existing tables.  You can use this interface to add, edit and remove custom tables.  By default, several custom tables have been created automatically by Tripal modules.  These tables hold data managed by the modules.  Also, tables used as materialized views appear in this list as well. They are identified with the `Is Mview` column.
```

File: docs/user_guide/custom_tables.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  You should not remove or edit custom tables that were created programmatically by tripal modules as it may cause problems with the proper function of the modules that created them.
+
+If you need to create a new table you should follow the following procedure.
+
+1.  Plan and design your table. It is best if you follow Chado design style so that your table would easily fit into Chado.
+2.  Reach out to the Chado community on the `Chado mailing list <https://sourceforge.net/projects/gmod/lists/gmod-schema>`_ and ask for advice on your design.
+3.  If you think your new table would benefit others, consider requesting it be added to a future version of Chado by adding an issue in the `Chado GitHub issue queue <https://github.com/gmod/chado/issues>`_.
+4.  Write your table in the `Drupal Schema API <https://www.drupal.org/docs/7/api/schema-api/introduction-to-schema-api>`_ array format.
+5.  Navigate to the  **Tripal → Data Storage → Chado -> Custom Tables** page on your site click the `Add Custom Table` link at the top and cut-and-paste the schema array and click the `Add` button.
```
Comment (bradfordcondon) [inline docs/user_guide/custom_tables.rst]:
```suggestion
5.  Navigate to the  **Tripal → Data Storage → Chado -> Custom Tables** page on your site. Click the `Add Custom Table` link at the top and cut-and-paste the schema array and click the `Add` button.
```

**Review by bradfordcondon — APPROVED** [review]
Guide looks good, nice and simple, thanks Stephen.

Not related to this PR but... what is the use case for this?  If you have to define your custom table writing a hook_schema compatible array either way, why not define it in a module so you can bundle the loader functionality with it?  It's for the programming averse, right (and thats why its in the user's guide not the dev's guide, makes sense)?

I'm glad you have the warning notice about not editing modules' custom tables.

**Comment by spficklin** [comment]
Thanks @bradfordcondon for the review!  Yes, you're right, the use case is if someone doesn't have the skills to develop a module to programmatically add the table. They can still use the bulk loader to get data into it.   We do need an equivalent document for programmatically creating tables.

**Pull request merged**